web/docs/language/features.md

```diff
@@ -178,138 +178,247 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
+```js title=ext/queries.js
 import HttpError from '@wasp/core/HttpError.js'
 
-export getTasks = async (args, context) => {
+export const getTasksError = async (args, context) => {
   const statusCode = 403
-  const message = 'You can\'t do this!'.
+  const message = 'You can\'t do this!'
   const data = { foo: 'bar' }
   throw new HttpError(statusCode, message, data)
 }
 ```
 
-and then in client it will be thrown as an Error with corresponding `.message` and `.data` fields (if status code is 4xx - otherwise `message` and `data` will not be forwarded to the client, for security reasons).
-
-This ensures that no error will accidentally leak out from the server, potentionally exposing sensitive data or implementation details.
+If the status code is `4xx`, the client will receive a response object with the corresponding `.message` and `.data` fields.
+To prevent information leakage, the server won't forward these fields for any other HTTP status codes.
 
-#### Using entities
-Most often, resources used by Operations will be Entities.
+#### Using Entities in Queries
+In most cases, resources used in Queries will be [Entities](#entity).
+To use an Entity in your Query, add it to the declaration in Wasp:
 
-To use an Entity in your Operation, declare in Wasp that Operation uses it:
-```c {4} title="todoApp.wasp"
+```c {4,9} title="main.wasp"
 // ...
-query getTasks {
-  fn: import { getAllTasks } from "@ext/foo.js",
+query fetchAllTasks {
+  fn: import { getAllTasks } from "@ext/queries.js",
+  entities: [Task]
+}
+
+query fetchFilteredTasks {
+  fn: import { getFilteredTasks } from "@ext/queries.js",
   entities: [Task]
 }
 ```
 
-This will inject specified entity into the context of your Operation.
-Now, you can access Prisma API for that entity like this:
-```js title="ext/foo.js"
+Wasp will inject the specified Entity into the Query's context argument, giving you access to the Entity's Prisma API:
+```js title="ext/queries.js"
 // ...
-export getAllTasks = async (args, context) => {
+
+export const getAllTasks = async (args, context) => {
   return context.entities.Task.findMany({})
 }
+
+export const getFilteredTasks = async (args, context) => {
+  return context.entities.Task.findMany({ 
+    where: { isDone: args.isDone }
+  })
+}
 ```
 
-where `context.entities.Task` actually exposes `prisma.task` from [Prisma API](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/crud).
+The object `context.entities.Task` exposes `prisma.task` from [Prisma's CRUD API](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/crud).
 
-#### Cache invalidation
-One of the trickiest part of managing web app state is making sure that data which queries are showing is up to date.
 
-Since Wasp is using react-query for managing queries, that means we want to make sure that parts of react-query cache are invalidated when we know they are not up to date any more.
+### Action
```

Comment:
Since the "Cache Invalidation" section refers to both Actions and Queries, I flipped the ordering to provide the necessary context.